bazel/re2c.bzl

```diff
@@ -0,0 +1,81 @@
+# re2c generator rules - assuming that re2c is available as '@re2c//:re2c'.
+
+visibility("public")
+
+def _re2c_impl(ctx):
+    srcs = ctx.files.srcs
+    if len(srcs) != 1:
+        fail("Attribute `srcs` must contain exactly 1 '.re2c' file.")
+    outs = ctx.outputs.outs
+    if not outs or [o for o in outs if o.path.split(".")[-1] not in ["c", "cc", "h"]]:
+        fail("Attribute `outs` must contain at least 1 '.cc' or '.h' file.")
+    err_file = ctx.actions.declare_file(ctx.attr.name + ".err")
+    ctx.actions.run_shell(
+        outputs = outs + [err_file],
+        tools = [ctx.executable._re2c_tool],
+        inputs = ctx.files.srcs + ctx.files.data,
+        command = """
+            line_args=""
+            [ -z "${re2c_parse_first_line}" ] && \\
+                line_args="$(sed -rne "1s,// re2c .INPUT -o .OUTPUT (.+),\\1,p" "{src}")"
```

Comment:
I believe this was needed for tests. Is it still needed?

If we want to provide a mechanism to specify re2c command-line options from the first line of a file, I think it should be done in a generic way by adding another re2c option, not by build system hacks. Most of the options already have a corresponding configuration anyway that can be specified in the file.

Reply:
Good point. That would be much better anyway - I might send a change for that.